### PR TITLE
Add imgcodec compatibility tests

### PR DESCRIPTION
**Pull request by szkarpinski**
## Category:

**New feature** (*non-breaking change which adds functionality*)

## Description:

This PR adds an imgcodec test that tests parsers for compatibility with old implementation.

A similar functionality for comparing with ImageMagick was extracted to https://github.com/NVIDIA/DALI/pull/4058 and will live as a standalone tool.

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [x] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2892

### CODE REVIEW

File: dali/imgcodec/image_format_test.cc [hunk 1]
```diff
@@ -75,6 +77,50 @@ class ImageFormatTest : public ::testing::Test {
...
+        }
+      }
+    }
+
+    if (nimages == 0)
+      FAIL() << "No matching images in " << directory;
+  }
+};
+
+class CompatibilityTest : public ComparisonTestBase {
```
Comment (jantonguirao) [inline dali/imgcodec/image_format_test.cc:110]: Can you please add doxygen saying what this tests does (comparing against the previous implementation)?

File: dali/imgcodec/image_format_test.cc [hunk 2]
```diff
@@ -134,6 +180,31 @@ TEST_F(ImageFormatTest, ReadHeaderStream) {
...
+}
+
+TEST_F(CompatibilityTest, DISABLED_Jpeg2000) {
+  RunOnDirectory(testing::dali_extra_path() + "/db/single/jpeg2k/", "jpeg2000", {".jp2"});
+}
+
+TEST_F(CompatibilityTest, DISABLED_WebP) {
+  RunOnDirectory(testing::dali_extra_path() + "/db/single/webp/", "webp", {".webp"});
+}
+
```
Comment (jantonguirao) [inline dali/imgcodec/image_format_test.cc:207]: where is JPEG?

Comment (szkarpinski) [reply dali/imgcodec/image_format_test.cc:207]: ¯\_(ツ)_/¯

File: dali/imgcodec/image_format_test.cc (hunk 1 above)
Comment (szkarpinski) [reply dali/imgcodec/image_format_test.cc:110]: Done

**Comment by szkarpinski** [comment]
!build

**Review by banasraf — APPROVED** [review]

**Comment by szkarpinski** [comment]
!build

**Pull request merged**